### ISSUE TITLE
bug(inference): Metal MTP/self-spec test oracles use last-wins argmax, contradicting the engine's first-wins contract

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#806]
> _Issue authored by Claude (Anthropic agent) on behalf of @ohdearquant._

## What

`crates/inference/src/forward/metal_qwen35.rs` contains two private test-only `argmax` helpers, one in the MTP test module and one in the self-speculative-decode test module (near lines 28437 and 29333). Both are implemented as:

```rust
fn argmax(logits: &[f32]) -> u32 {
    logits
        .iter()
        .copied()
        .enumerate()
        .max_by(|(_, a), (_, b)| a.partial_cmp(b).unwrap_or(std::cmp::Ordering::Equal))
        .map(|(i, _)| i as u32)
        .unwrap_or(0)
}
```

`Iterator::max_by` returns the **last** element when multiple elements compare equal. This contradicts the production/reference tie-break contract in `crates/inference/src/speculative.rs`, whose public `argmax` (around line 107) explicitly documents and implements **first-wins** on ties:

```rust
/// When multiple elements share the maximum value, returns the **first**
/// occurrence. Speculative decoding must be token-identical to plain greedy...
pub fn argmax(logits: &[f32]) -> usize {
    let mut best_idx = 0;
    let mut best_val = f32::NEG_INFINITY;
    for (i, &v) in logits.iter().enumerate() {
        if v > best_val {
            best_val = v;
            best_idx = i;
        }
    }
    best_idx
}
```

Both test-only helpers back oracle functions (`plain_greedy_oracle` and its self-spec counterpart) that these test modules use as the ground truth to check MTP/self-speculative decoding token-for-token agreement. A last-wins oracle can silently mask a first-wins regression in the production Metal decode path whenever two logits tie exactly (a real occurrence in low-precision/greedy edge cases and in constructed test fixtures with repeated logit values).

## Why

The test oracle's job is to prove that speculative decoding is token-identical to plain greedy decoding. If the oracle itself uses a different tie-break rule than the thing it's meant to model (`crate::speculative::argmax`, which the actual engine calls), the oracle is not testing the real contract — it is testing an incompatible one. This is a correctness gap in test infrastructure, not a production bug, and should be fixed before any deletion work touches this file so that deletion PRs remain behavior-pure and this fix's oracle-correctness change stays isolated and reviewable on its own.

## Steps

1. In `crates/inference/src/forward/metal_qwen35.rs`, delete both private `argmax` helper functions (in the MTP test module and the self-spec test module).
2. Replace their call sites with `crate::speculative::argmax(logits)`, casting the returned `usize` to `u32` at the call site (the helpers currently return `u32`; `crate::speculative::argmax` returns `usize`).
3. Add an explicit tie-break regression test to both modules: a logits vector such as `[1.0, 1.0, 0.0]` must resolve to index `0` (first-wins), not index `1`.
4. Do not change any other production code path in this PR — this is a test-oracle-only fix.

## Gates

- Named CPU-only tests pass in both the MTP test module and the self-spec test module (no GPU/Metal hardware required for these specific oracle tests).
- Reverse-apply this fix (restore the `max_by`-based helpers) and confirm the new tie-break assertions fail — this proves the regression test is mutation-sensitive, not decorative.
- Full existing `cargo test` suite for `lattice-inference` passes unaffected.
- Semver class: none (test-only code, no public API surface touched).

## Sequencing

- **Depends on:** nothing; this is a standalone correctness fix and should land before or independently of any deletion PR touching `crates/inference/src/generate.rs` or `crates/inference/src/forward/batch_prefill.rs`.
- **Depended on by:** none directly, but should land before or alongside the 0.6.0 removal work in this cluster so the test suite's oracle correctness isn't conflated with unrelated deletions in the same review.
